src/test.cpp

```diff
@@ -0,0 +1,382 @@
+        // run motors in negative direction for 2 seconds (encoder count should decrease)
+        long startTime = millis(); 
+        long theta0 = encoder.read();
+
+        Serial.println("Starting motor/encoder direction test...");
+
+        speed = 250;
+        utility::runMotor(speed); //test for new class
+        while (millis() - startTime < 1000) {}
+        long theta1 = encoder.read();
+        String msg = ((theta1-theta0) > 0) ? "\tPASS":"\tFAIL";
+        Serial.println("Running motors + direction. t0, t1" + String(theta0) + "\t" +  String(theta1) + msg);
+        startTime = millis();
+        speed = -speed;
+        utility::runMotor(speed);
+        while (millis() - startTime < 1000) {}
+        long theta2 = encoder.read();
+        msg = ((theta2-theta1) < 0) ? "\tPASS":"\tFAIL";
+        Serial.println("Running motors + direction. t1, t2: " + String(theta1) + "\t" + String(theta2) + msg);
+        speed = 0;
+        utility::runMotor(speed);
+    }
+
+//if anything breaks it might be because I stored 
+//the common values of speed, motorAngle etc in common class variables
+
+    void motorPowerTest() {
+        String inString = "";
+        // int speed = 0;
+        unsigned long lastPrint = 0;
+        long angle = 0;
+
+        while (true){
+            utility::runMotor(speed);
+
+            if (millis() - lastPrint > 200){
+                angle = encoder.read();
+                Serial.println(String(speed)+"\t"+String(angle));
+                lastPrint = millis();
+            }
+
+            while (Serial.available() > 0) {
+                //Read incoming commands
+                int inChar = Serial.read();
+                
+                if (inChar == '\n') {
+                    if (inString == "fin"){
+                        return;
+                    }else{
+                        speed = inString.toInt();
+                    }
+                    inString = "";
+                } else {
+                    inString += (char)inChar;
+                }
+                
+            }
+
+        }
+    }
+
+    void ptTest() {
+        unsigned long lastPrint = 0;
+        String inString = "";
+        // print PT reading 6 times, at 0.5s intervals
+        #ifndef USE_DASHBOARD
+        Serial.println("Starting PT test...");
+        #endif
+        long p;
+        long old_p;
+        int count = 0;
+        Buffer p_buff(BUFF_SIZE);
+        long t;
+        while (true) {
+            count++;
+            double pressure = utility::voltageToPressure(analogRead(LP_PT));
+            t = micros();
+            p_buff.insert(double(t)/1.0e6, pressure);
+            double old_pressure = pressure;
+            long old_t = t;
+            
+            if (millis() - lastPrint > 100) {
+                
+                lastPrint = millis();
+                count = 0;
+
+                #ifndef USE_DASHBOARD
```

Comment:
USE_DASHBOARD is defined in ElectronicRegulator.cpp, so even if we set that flag because we want dashboard to be used, this part will never be compiled.